x/concentrated-liquidity/pool.go

```diff
@@ -340,3 +343,45 @@ func validateAuthorizedQuoteDenoms(ctx sdk.Context, denom1 string, authorizedQuo
 func (k Keeper) GetLinkedBalancerPoolID(ctx sdk.Context, concentratedPoolId uint64) (uint64, error) {
 	return k.gammKeeper.GetLinkedBalancerPoolID(ctx, concentratedPoolId)
 }
+
+func (k Keeper) GetUserUnbondingPositions(ctx sdk.Context, address sdk.AccAddress) ([]model.PositionWithPeriodLock, error) {
+	// Get the position IDs for the specified user address.
+	positions, err := k.GetUserPositions(ctx, address, 0)
+	if err != nil {
+		return nil, err
+	}
+
+	// Query each position ID and determine if it has a lock ID associated with it.
+	// Construct a response with the position as well as the lock's info.
+	var userPositionsWithPeriodLocks []model.PositionWithPeriodLock
+	for _, pos := range positions {
+		lockId, err := k.GetLockIdFromPositionId(ctx, pos.PositionId)
+		switch err.(type) {
+		case types.PositionIdToLockNotFoundError:
+			continue
+		case nil:
+			// If we have hit this logic branch, it means that, at one point, the lockId provided existed. If we fetch it again
+			// and it doesn't exist, that means that the lock has matured.
+			lock, err := k.lockupKeeper.GetLockByID(ctx, lockId)
+			if err == errorsmod.Wrap(lockuptypes.ErrLockupNotFound, fmt.Sprintf("lock with ID %d does not exist", lock.GetID())) {
+				continue
+			}
+			if err != nil {
+				return nil, err
+			}
+
+			// Don't include locks that aren't unlocking
+			if lock.EndTime.IsZero() {
+				continue
+			}
+
+			userPositionsWithPeriodLocks = append(userPositionsWithPeriodLocks, model.PositionWithPeriodLock{
+				Position: pos,
+				Locks:    *lock,
+			})
+		default:
+			continue
+		}
```

Comment:
Similar to the comment in another PR, why do we have a default case here?

Reply:
https://github.com/osmosis-labs/osmosis/pull/5684/commits/6715ba8e5876cad3229e030e12dcfc27dd75ba63